Neos.ContentRepository.Core/Classes/SharedModel/ContentRepository/ContentRepositoryIds.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/*
+ * This file is part of the Neos.ContentRepository package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+declare(strict_types=1);
+
+namespace Neos\ContentRepository\Core\SharedModel\ContentRepository;
+
+/**
+ * @implements \IteratorAggregate<ContentRepositoryId>
+ * @api
+ */
+final readonly class ContentRepositoryIds implements \IteratorAggregate, \Countable
+{
+    /**
+     * @var array<ContentRepositoryId>
+     */
+    private array $ids;
+
+    private function __construct(ContentRepositoryId ...$ids)
+    {
+        $this->ids = $ids;
+    }
+
+    /**
+     * @param array<ContentRepositoryId|string> $ids
+     */
+    public static function fromArray(array $ids): self
+    {
+        $processedIds = [];
+        foreach ($ids as $id) {
+            if (is_string($id)) {
+                $id = ContentRepositoryId::fromString($id);
+            }
+            if (!$id instanceof ContentRepositoryId) {
```

Comment:
i kinda prefer using language level varadics here to do the type check - as that must be faster - and by the doc comment `@param array<ContentRepositoryId|string>` this case may actually never occur.

but still ... just nitpicking ^^

Reply:
I just claim boldly: Performance isn't an issue here.
Even if you had 1000 Content Repositories in one installation, this check would cost nothing basically.
I added this because it just provides a nicer exception message, e.g. when you configure your content repositories incorrectly